examples/acoustics/acoustic.py

```diff
@@ -0,0 +1,179 @@
+
+import os
+import argparse
+
+import numpy as np
+import pandas as pd
+from matplotlib import pyplot as plt
+import matplotlib.ticker as ticker
+
+import flow360client
+
+def getCaseConfigParameters(caseId):
+    return flow360client.case.GetCaseInfo(caseId)['runtimeParams']
+
+def readAcousticCSVFile(acsv):
+    # read acoustic csv when the file is available via python api
+    # with tempfile.TemporaryDirectory() as tmpdir:
+    #     tmpfile = os.path.join(tmpdir, 'aeroacoustics_v3.csv')
+    #     flow360client.case.DownloadFile(caseId, 'aeroacoustics_v3.csv', tmpfile)
+    pressure_data = pd.read_csv(acsv,skipinitialspace=True)
+    times = pressure_data['physical_time']
+    steps = pressure_data['physical_step']
+    p = pressure_data.iloc[:,2:-1]
+    return [np.array(times),np.array(steps),np.array(p)]
+
+def filterZeroAcousticData(time,steps,pressure):
+    numObservers = pressure.shape[-1]
+    acoustic_data_indices = {}
+    acoustic_data = {}
+    # loop over observers to find where they are zero
+    for i in range(numObservers):
+        p_observer = pressure[:,i]
+        acoustic_data_indices[i] = np.where(p_observer!=0)[0]
+    first_elements = []
+    last_elements = []
+    # lists of elements with zeros at begin and end
+    for i in range(numObservers):
+        first_elements.append(acoustic_data_indices[i][0])
```

Comment:
You need to handle the case where the data is non zero over all time steps or only has zero entries in the beginning or end.

You can try the numpy trim_zeros function, but that does not return index

Reply:
The way it is, it handles data when they are only zero at the beginning or end.

I made a modification to handle all-zero data for an observer. 

Please use example2_aeroacoustics_v3.csv for these scenarios.